builder/docker_test.go

```diff
@@ -0,0 +1,170 @@
+package builder
+
+import (
+	// "fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func TestGetDirFromConfigPath(t *testing.T) {
+	tests := []struct {
+		Filepath        string
+		ExpectedDirPath string
+		ShouldFail      bool
+	}{
+		{
+			Filepath:        "some/path/to/directory",
+			ExpectedDirPath: "some/path/to/directory",
+		},
+		{
+			Filepath:        "some/path/to/file",
+			ExpectedDirPath: "some/path/to",
+		},
+		{
+			Filepath:        "some/path/to/doesnotexist",
+			ExpectedDirPath: "some/path/to",
+		},
+
+		// Error cases.
+		{Filepath: "not/a/real/path", ShouldFail: true},
+	}
+
+	testDir, err := ioutil.TempDir("", "getdirfrompath-test-")
+	if err != nil {
+		t.Fatalf("couldn't create temporary directory to write test cases: %s", err)
+	}
+	defer func() {
+		_ = os.RemoveAll(testDir)
+	}()
+
+	// Set up the dirs/files for testing
+	path := filepath.Join(testDir, "some/path/to/directory")
+	if err = os.MkdirAll(path, 0777); err != nil {
+		t.Errorf("Failed to create test directory: %q\n%s", path, err)
```

Comment:
Should be `Fatal` no? IMO test setup failures should be fatal, while actual test failures can be just `Error`s

Reply:
Good point -- I agree. I copied the bundleset tests as my starting point, so at least a couple other places are doing it this way. I won't change them as part of this, but that'd be a good cleanup task down the line.